spec/controllers/global_search_controller_spec.rb

```diff
@@ -168,7 +168,12 @@ def it_should_have_customer_paths
       end
 
       context "when there is an order and order detail with same ids" do
-        let!(:order2) { FactoryGirl.create(:purchased_order, id: order_detail.id, product: product) }
+        let!(:order2) do
+          # random test failure, if by chance there is an order with the same ID
+          # in the spec suite.
+          Order.find_by(id: order_detail.id) || FactoryGirl.create(:purchased_order, id: order_detail.id, product: product)
```

Comment:
I need to think through this, but I'm not quite convinced this is correct. Either way, can we pull this out into a separate PR? It doesn't really belong with the nokogiri upgrade.